cmd/sst-ctl/main.go

```diff
@@ -171,3 +173,226 @@ func subCmdBF(args []string) error {
+	var infomap map[int]*sst.SstPackageInfo
+	var err error
+	var priority int
+
+	defaultMaxFreq := 0xff
+
+	flags := flag.NewFlagSet("cp", flag.ExitOnError)
+	flags.BoolVar(&enable, "enable", false, "Enable feature")
+	flags.BoolVar(&disable, "disable", false, "Disable feature")
+	flags.BoolVar(&reset, "reset", false, "Reset CP to a known state")
+	flags.IntVar(&clos, "clos", -1, "Class of service (0 - 3)")
+	flags.IntVar(&epp, "epp", 0, "Energy Performance Preference value, Lower value favors performance, and higher value favors power. The value can be between 0 and 100. The default value is 0.")
+	flags.IntVar(&minFreq, "min", 0, "Clos minimum frequency MHz")
+	flags.IntVar(&maxFreq, "max", defaultMaxFreq, "Clos maximum frequency MHz")
+	flags.IntVar(&desiredFreq, "desired", 0, "Clos desired frequency MHz")
+	flags.IntVar(&proportionalPriority, "proportional", 0, "Clos proportional priority weight. Used if CP priority mode is 0 (Proportional)")
+	flags.IntVar(&priority, "priority", 1, "CP priority mode. 0 is Proportional, 1 is Ordered.")
+	flags.StringVar(&cpuStr, "cpus", "", "List of CPUs assigned to the Clos.")
+
+	addGlobalFlags(flags)
+
+	flags.Usage = func() {
+		flags.PrintDefaults()
+
+		fmt.Fprintf(os.Stderr, "\nExample usage:\n\n")
+		fmt.Fprintf(os.Stderr, "First reset CP to default:\n\t%s cp -reset\n\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "Then set the CLOS values:\n\t%s cp -clos 1 -desired 280 -epp 1 -max 30 -min 21 -priority 1 -package 0\n\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "Then bind CPUs to a CLOS:\n\t%s cp -clos 1 -cpus 1,3,5,6\n\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "Finally enable CP:\n\t%s cp -enable -package 0\n\n", os.Args[0])
+	}
+
+	if err := flags.Parse(args); err != nil {
+		return err
+	}
+
+	if reset {
+		err := sst.ResetCPConfig()
+		_ = printPackageInfo()
+		return err
+	}
+
+	if enable && disable {
+		fmt.Printf("Please provide either -enable or -disable flag\n")
+		return fmt.Errorf("Please provide either -enable or -disable flag")
+	}
+
+	if minFreq < 0 {
+		fmt.Printf("Too low value %d for min frequency\n", minFreq)
+		return fmt.Errorf("Invalid value %d for min freq", minFreq)
+	}
+
+	if epp < 0 {
+		fmt.Printf("Too low value %d for EPP, using 0 instead\n", epp)
+		epp = 0
+	}
+
+	if epp > 100 {
+		fmt.Printf("Too high value %d for EPP, using 100 instead\n", epp)
+		epp = 100
+	}
+
+	// If user specifies a list of CPUs, then he/she wants to assign those
+	// CPUs to a specific CLOS. If the -cpus option is not set, then user
+	// wants to configure the actual CLOS values. Both operations cannot be
+	// done at the same time.
+
+	// If user specifies a list of CPUs, then the package option is ignored.
+	// Verify that all the CPUs belong to one specific package.
+	if cpuStr != "" {
+		cpus = utils.NewIDSet(str2slice(cpuStr)...)
+
+		infomap, info, pkgs, err = getPackage(packageIds, cpus)
+		if err != nil {
+			fmt.Printf("Cannot get CPUs %v package\n", cpus)
+			return err
```

Comment:
nit: I thinki the printf is redundant.

You can wrap `err` in a more informative error msg if needed

```suggestion
			return fmt.Errorf("Cannot get CPUs %v package: %w, cpus, err)
```